File: src/leetcode/problem_342/Solution.java
package leetcode.problem_342;

public class Solution {
    public boolean isPowerOfFour(int n) {
        if (n == 0 || Integer.bitCount(n) > 1) return false;
        int comparator = 1;
        for (int i = 1; i <= 16; i++) {
            if (n == comparator) return true;
            comparator = comparator << 2;
        }
        return false;
    }
}

File: src/leetcode/problem_342/solution.kt
package leetcode.problem_342

import leetcode.problem_231.isPowerOfTwo

fun main() {
    println(isPowerOfTwo(0))
    println(isPowerOfFour(4))
    println(isPowerOfFour(16))
    println(isPowerOfFour(64))


}

fun isPowerOfFour(n: Int): Boolean {
    if (n == 0) return true
    if (n.countOneBits() != 1) return false
    var comparator = 1
    for (i in 1..16) {
        if (n == comparator) return true
        comparator = comparator shl 2
    }
    return false
}